### ISSUE TITLE
Event processing order

### ISSUE THREAD

**Issue by carlosper88** [boostercloud/booster#370]
I have a doubt about the order in which the events will be handled/reduced.

Let's say I send a command with `CreateUser`. That command will then register an event `UserCreated` that will be reduced by a `User` entity. 

In my application, I want to be able to access that `User` that has just been created (via Booster.fetchEntitySnapshot), so I create a new event `AfterUserCreation` that will have `User` as a param. This new event will be fired by the `UserCreated` event handler:

```
@EventHandler(UserCreated)
export class HandleUserCreated {
    public static async handle(
        event: UserCreated,
        register: Register
    ): Promise<void> {

       //Here is where I'd perform Booster.fetchEntitySnapshot to get User

        register.events(new AfterUserCreation(
           //Here I'd pass the User resulting from the fetch
        ))
    }
}
```

My main concern here is whether the event handler will be able to fetch the `User` entity. I mean, WHEN does the event handler comes into action? does it wait until the `User`entity reduces the first `UserCreated`? or is it launched "in parallel"?

**Comment by NickSeagull** [comment]
In the case of the AWS provider, the event handlers are subscribed to the change stream from the event store, which is DynamoDB. This is the same case for the reducers, so in your example, you could say that they both are launched in parallel.

**Comment by NickSeagull** [comment]
Closing, feel free to reopen if you have more questions on this topic

**Comment by javiertoledo** [comment]
I have one question about the semantics in your use case. If your `User` is created with the data in `UserCreated` exclusively, why do you need to "capture" the state of `User` and return it in a new event called `AfterUserCreation`? Wouldn't it be easier that you capture the `UserCreated` event from wherever you're using this new `AfterUserCreation` event and use the data in `UserCreated` without checking the actual entity value?

Once an event is written in the event-store, we consider the operation done, you don't need to wait for the entity to be snapshotted to consider it "created", snapshots are just a way to improve read performance.

**Comment by carlosper88** [comment]
@javiertoledo 
In my case, I needed a second event because the entity in charge of reducing the second event was not the `User` but another one. 
At first, I tried to reduce the `UserCreated` event inside two different entities, `User` and another one called `UserContacts` (I wanted the previous `UserContacts entity for some checkups), but I saw it was not possible (error: `Error registering reducer: The event UserCreated was already registered to be reduced by method create in the entity User`), so that's why I tried this approach.

Anyway, current Booster implementation does not support filtering while doing `Booster.fetchEntitySnapshot` so the feature I wanted to implement will have to wait a bit.
